exporter/kinesisexporter/exporter.go

```diff
@@ -16,57 +16,90 @@ package kinesisexporter
 
 import (
 	"context"
+	"fmt"
 
-	kinesis "github.com/signalfx/opencensus-go-exporter-kinesis"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	jaegertranslator "go.opentelemetry.io/collector/translator/trace/jaeger"
 	"go.uber.org/zap"
 )
 
-// Exporter implements an OpenTelemetry trace exporter that exports all spans to AWS Kinesis
+const (
+	errInvalidContext = "invalid context"
+)
+
+// Exporter implements an OpenTelemetry exporter that pushes OpenTelemetry data to AWS Kinesis
 type Exporter struct {
-	kinesis *kinesis.Exporter
-	logger  *zap.Logger
+	producer producer
+	logger   *zap.Logger
 }
 
-var _ component.TraceExporter = (*Exporter)(nil)
+// newExporter creates a new exporter with the passed in configurations.
+// It starts the AWS session and setups the relevant connections.
+func newExporter(c *Config, logger *zap.Logger) (*Exporter, error) {
+	pr, err := newKinesisProducer(c, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{producer: pr, logger: logger}, nil
+}
 
 // Start tells the exporter to start. The exporter may prepare for exporting
 // by connecting to the endpoint. Host parameter can be used for communicating
-// with the host after Start() has already returned. If error is returned by
-// Start() then the collector startup will be aborted.
-func (e Exporter) Start(_ context.Context, _ component.Host) error {
+// with the host after start() has already returned. If error is returned by
+// start() then the collector startup will be aborted.
+func (e *Exporter) Start(ctx context.Context, _ component.Host) error {
+	if ctx == nil || ctx.Err() != nil {
+		return fmt.Errorf(errInvalidContext)
+	}
+
+	e.producer.start()
 	return nil
 }
 
-// Shutdown is invoked during exporter shutdown.
-func (e Exporter) Shutdown(context.Context) error {
-	e.kinesis.Flush()
+// Shutdown is invoked during exporter shutdown
+func (e *Exporter) Shutdown(ctx context.Context) error {
+	if ctx == nil || ctx.Err() != nil {
+		return fmt.Errorf(errInvalidContext)
+	}
```

Comment:
In case context is cancelled or hits deadline, do we want to exit without stopping? I don't think we use it right now but one can imagine the collector add some sort of deadline on the context to give all components a few seconds to cleanup and if they don't then force close. Is this why we exit ASAP in this case or are there other scenarios?